docs/development/adding-a-package-manager.md

```diff
@@ -67,3 +77,16 @@ This is used when more than one package manager shares settings from a common la
 ### `supportsLockFileMaintenance` (optional)
 
 Set to true if this package manager needs to update lock files in addition to package files.
+
+### `updateArtifacts` (async, optional)
+
+This function is triggered if a dependency has been updated for a package file or `lockfileMaintenance` has been triggered.
+Mostly this function is used to directly or indirectly ( e.g. by executing external binaries ) update lock files.
```

Comment:
```suggestion
Mostly this function is used to directly (or indirectly) update lock files.
An example of indirectly updating the lock file: running external binaries.
```

How about something like this? I'm trying to split the paragraph into smaller parts.